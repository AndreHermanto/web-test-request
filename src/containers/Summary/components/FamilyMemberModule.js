import React, { Component } from 'react';
import { 
  Row, 
  Col,
  Glyphicon,
  Button 
} from 'react-bootstrap';
import { 
  SummaryBox,
  SummaryHeading,
  SummaryBreakLine,
  SummaryTitle,
  SummaryDetails,
  SummaryNotes,
  SectionHeading,
  Tag
} from '../summaryStyled'; 

/**
* FamilyMemberModule summary - UI for summary page to display all family member details/clinical info data.
*/
class FamilyMemberModule extends Component
{
  constructor(props) {
    super(props);
    this.handleOnClick = this.handleOnClick.bind(this);
  }

  handleOnClick() {
    this.props.handleOnClick('step4');
    return 4;
  }

  render() {
    return(
      <SummaryBox>
        <SummaryHeading> 
          <Row>
            <Col md={12}>
              Family Member ({this.props.familyMemberDetails.relationship}) - {this.props.familyMemberDetails.firstName + ' ' + this.props.familyMemberDetails.lastName}
            </Col>
            <Col md={12}>
              <Button 
                bsStyle="link"
                onClick={this.handleOnClick}
              >
                <Glyphicon glyph="pencil"/> Edit
              </Button>
              <Tag bsStyle={this.props.familyMemberClinicalInfo.affected ? 'danger' : 'success'}>
                {this.props.familyMemberClinicalInfo.affected ? 'Affected' : 'Unaffected'}
              </Tag>
            </Col>
          </Row>
        </SummaryHeading>
        <SummaryBreakLine/>
        <Row>
          <Col md={12}>
            <SectionHeading> 
              Details
            </SectionHeading>
          </Col>
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Date of Birth:  </SummaryTitle>
              {this.props.familyMemberDetails.dob}
            </SummaryDetails>
          </Col>
          {
            this.props.familyMemberDetails.medicalRecordNo !== '' && 
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Medical record number:  </SummaryTitle>
                  {this.props.familyMemberDetails.medicalRecordNo}
                </SummaryDetails>
            </Col>
          }
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Gender:  </SummaryTitle>
              {
                this.props.familyMemberDetails.gender === 'Other' ?
                this.props.familyMemberDetails.genderOther + ' (Other)' :
                this.props.familyMemberDetails.gender
              } 
            </SummaryDetails>
          </Col>
          {
            this.props.familyMemberDetails.ethnicity !== '' && 
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Ethnicity:  </SummaryTitle>
                {this.props.familyMemberDetails.ethnicity}
              </SummaryDetails>
            </Col>
          }
          {
            this.props.familyMemberDetails.deceased !== false &&
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Deceased:  </SummaryTitle>
                {this.props.familyMemberDetails.deceased ? 'Yes' : 'No'}
              </SummaryDetails>
            </Col>
          }
          {
            (this.props.familyMemberDetails.deceased !== false && this.props.familyMemberDetails.sampleSource !== '') && 
            <Col md={12}>
              <SummaryDetails> 
                <SummaryTitle> Sample source:  </SummaryTitle>
                {this.props.familyMemberDetails.sampleSource}
              </SummaryDetails>
            </Col>
          }
          <Col md={12}>
            <SummaryDetails> 
              <SummaryTitle> Email:  </SummaryTitle>
              {this.props.familyMemberDetails.email}
            </SummaryDetails>
          </Col>
          <Col md={12}>
            <SectionHeading> 
              Clinical Information
            </SectionHeading>
          </Col>
          <Col md={12}>
            <SummaryTitle> Clinical note </SummaryTitle>
            <SummaryNotes>
              {this.props.familyMemberClinicalInfo.clinicalInfo}
            </SummaryNotes>
          </Col>
          {
            this.props.familyMemberClinicalInfo.relevantInvestigation !== '' &&
            <Col md={12}>
              <SummaryTitle> Relevant investigation note </SummaryTitle>
              <SummaryNotes>
                {this.props.familyMemberClinicalInfo.relevantInvestigation}
              </SummaryNotes>
            </Col>
          }
          {
            this.props.familyMemberClinicalInfo.familyHistory !== '' &&
            <Col md={12}>
              <SummaryTitle> Family history note </SummaryTitle>
              <SummaryNotes>
                {this.props.familyMemberClinicalInfo.familyHistory}
              </SummaryNotes>
            </Col>
          }
          <Col md={12}>
            <SummaryDetails> 
            <SummaryTitle> Consanguinity:  </SummaryTitle>
              {this.props.familyMemberClinicalInfo.consanguinity ? 'Yes' : 'No'} 
            </SummaryDetails>
          </Col>
        </Row>
      </SummaryBox>
    )
  }
}

export default FamilyMemberModule;