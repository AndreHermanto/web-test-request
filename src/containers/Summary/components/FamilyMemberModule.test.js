import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import FamilyMemberModule from './FamilyMemberModule';

function getStep(value) {
  return value;
}
describe('FamilyMemberModule test', () => {
  const member = {
      "FamilyMemberDetails":{"lastName":"abc@abc.com","firstName":"abc@abc.com","dob":"2-January-1918","medicalRecordNo":"abc@abc.com","gender":"Female","genderOther":"","ethnicity":"","deceased":false,"sampleSource":"","consent":false,"email":"abc@abc.com"},
      "FamilyMemberClinicalInfo":{"clinicalInfo":"abc@abc.com","affected":true,"relevantInvestigation":"asdasdasd","familyHistory":"asdasdasdasd","consangunity":false,     consanguinityInfo: "asjhdasbdlkahsdjb"}
  };  

  const props = {
    familyMemberDetails: member.FamilyMemberDetails,
    familyMemberClinicalInfo: member.FamilyMemberClinicalInfo,
    handleOnClick: getStep
  };
    
  test('renders without crashing ', () => {
    const page = renderer.create(React.createElement(FamilyMemberModule, props)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <FamilyMemberModule familyMemberDetails={member.FamilyMemberDetails} familyMemberClinicalInfo={member.FamilyMemberClinicalInfo}
              handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step4')).toEqual('step4');
    expect(page.handleOnClick()).toEqual(4);
  });
});
