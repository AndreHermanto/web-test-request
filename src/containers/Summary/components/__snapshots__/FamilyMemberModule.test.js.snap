exports[`FamilyMemberModule test renders without crashing  1`] = `
<div
  className="iKCvIz">
  <h3
    className="iArjKn">
    <div
      className="row">
      <div
        className="col-md-12">
        Family Member (
        ) - 
        abc@abc.com abc@abc.com
      </div>
      <div
        className="col-md-12">
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button">
          <span
            className="glyphicon glyphicon-pencil" />
           Edit
        </button>
        <span
          className="bQxwRk label label-danger">
          Affected
        </span>
      </div>
    </div>
  </h3>
  <div
    className="dgEelt" />
  <div
    className="row">
    <div
      className="col-md-12">
      <h4
        className="hWtWAB">
        Details
      </h4>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Date of Birth:  
        </span>
        1918/January/2
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Medical record number:  
        </span>
        abc@abc.com
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Gender:  
        </span>
        Female
      </p>
    </div>
    <div
      className="col-md-12">
      <h4
        className="hWtWAB">
        Clinical Information
      </h4>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Clinical note 
      </span>
      <div
        className="gczsYD">
        abc@abc.com
      </div>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Relevant investigation note 
      </span>
      <div
        className="gczsYD">
        asdasdasd
      </div>
    </div>
    <div
      className="col-md-12">
      <span
        className="bdMjbs">
         Family history note 
      </span>
      <div
        className="gczsYD">
        asdasdasdasd
      </div>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        <span
          className="bdMjbs">
           Consanguinity:  
        </span>
        No
      </p>
    </div>
  </div>
</div>
`;
